src/ir/context.rs

```diff
@@ -2412,6 +2428,70 @@ If you encounter an error missing from this list, please file an issue or a PR!"
         }
     }
 
+    /// Compute the items for dynamic generation.
+    fn compute_dynamic_generation_items(&mut self) {
+        assert!(self.in_codegen_phase());
+        assert!(self.current_module == self.root_module);
+        let _t = self.timer("compute_dynamic_generation_items");
+
+        let dyngen_items = self
+            .items()
+            // Only consider items that are enabled for codegen.
+            .filter(|&(_, item)| item.is_enabled_for_codegen(self))
+            .filter(|&(_, item)| {
+                // If the user has not chosen to do dynamic loading, then we have nothing to
+                // do.
+                if !self.options().dynamic_loading {
```

Comment:
Seems like this check should really go at the beginning of the function rather than being done for every item.

Reply:
Quite right -- will fix.